lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -280,11 +319,12 @@ def setup_access_token
         response['access_token']
       end
 
-      def commit(action, post, authorization = nil)
+      def commit(action, post, authorization = nil, method = :post)
         begin
-          raw_response = (action == :verify_payment ? ssl_get("#{base_url}/payments/#{post}", headers) : ssl_post(url(post, action, authorization), post.to_json, headers))
+          raw_response = ssl_request(method, url(action, authorization), post.nil? ? nil : post.to_json, headers(action))
```

Comment:
I think you might want `post.empty?` here. If `post` is already `nil`, then this would already resolve to `nil`, but if `post` is an empty hash `{}` then `post.nil?` would resolve to false and still pass a value.

Reply:
Instead of using post.empty? (because if we use `nil.empty?` it will return an error), I choose to use both; if it's not nil, then check if it's not empty.